### PR TITLE
Fix JWK kty from "rsa" to "RSA"

### PR DESCRIPTION
**Pull request by marshall-lee**
JWKS libraries expect it to be "RSA", not "rsa", example: https://github.com/auth0/node-jwks-rsa/blob/6cfa98f8ac045b7cd1dd07fd5ac01b00d82d60cf/src/JwksClient.js#L79-L81

According to RFCs, "kty" field seems to be case-sensitive, though there cannot be names matching in a case-insensitive manner: https://tools.ietf.org/html/rfc7518#section-7.4.1

### CODE REVIEW

**Review by a-palchikov — COMMENTED** [review]
Can you link in a comment to the RFC7518, [section 6.1](https://tools.ietf.org/html/rfc7518#section-6.1) where the values are listed for completeness?

File: lib/defaults/defaults.go [hunk 1]
```diff
@@ -615,7 +615,7 @@ const (
 
 const (
 	// ApplicationTokenKeyType is the type of asymmetric key used to sign tokens.
-	ApplicationTokenKeyType = "rsa"
+	ApplicationTokenKeyType = "RSA"
```
Comment (a-palchikov) [inline lib/defaults/defaults.go:618]: No, I would link to the RFC here, in the Doc comment:
```go
const (
	// ApplicationTokenKeyType is the type of asymmetric key used to sign tokens.
	// See https://tools.ietf.org/html/rfc7518#section-6.1 for possible values
	ApplicationTokenKeyType = "RSA"
)
```

**Review by a-palchikov — APPROVED** [review]

**Review by webvictim — APPROVED** [review]

**Comment by benarent** [comment]
@russjones Is it possible to get this back ported. 

I've been trying https://github.com/gravitational/teleport-plugins/pull/147 but it's not working due to this.  Latest Teleport Cloud instance is still showing `rsa`.

image

**Comment by a-palchikov** [comment]
Here's the [PR](https://github.com/gravitational/teleport/pull/5155) for the 5.x branch.

**Comment by benarent** [comment]
Thanks @a-palchikov !

**Pull request merged**